realm/src/main/java/io/realm/Realm.java

```diff
@@ -212,18 +216,20 @@ protected void finalize() throws Throwable {
      */
     @Override
     public void close() {
-        Map<String, Integer> localRefCount = referenceCount.get();
-        Integer references = localRefCount.get(canonicalPath);
+        Map<RealmConfiguration, Integer> localRefCount = referenceCount.get();
+        String canonicalPath = configuration.getPath();
+        Integer references = localRefCount.get(configuration);
         if (references == null) {
             references = 0;
         }
         if (sharedGroup != null && references == 1) {
-            realmsCache.get().remove(canonicalPath);
+            realmsCache.get().remove(configuration);
+            globalPathConfigurationCache.get(canonicalPath).remove(configuration);
             sharedGroup.close();
             sharedGroup = null;
-            AtomicInteger counter = openRealms.get(canonicalPath);
+            AtomicInteger counter = globalOpenInstanceCounter.get(canonicalPath);
```

Comment:
According to the doc https://github.com/realm/realm-java/blob/cm-configuration-builder-cache-compare/realm/src/main/java/io/realm/Realm.java#L146 , this is only needed by `deleteRealmFile` !?